core/src/main/scala/org/apache/spark/serializer/GenericAvroSerializer.scala

```diff
@@ -44,8 +44,8 @@ import org.apache.spark.util.Utils
  *                string representation of the Avro schema, used to decrease the amount of data
  *                that needs to be serialized.
  */
-private[serializer] class GenericAvroSerializer(schemas: Map[Long, String])
-  extends KSerializer[GenericRecord] {
+private[serializer] class GenericAvroSerializer[D >: Null <: GenericContainer]
```

Comment:
Why do we need the lower bound on `Null` here?

Reply:
We need to pass in a null at https://github.com/apache/spark/blob/735246119836f20811c67193f778246e1157e63e/core/src/main/scala/org/apache/spark/serializer/GenericAvroSerializer.scala#L155 where the compiler expects `D`. Without the lower bound, the Scala compiler won't let me do that.

Reply:
Fascinating, thanks!

Reply:
Done